src/GraphQL.MemoryCache/MemoryDocumentCache.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using GraphQL.Language.AST;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace GraphQL.Caching
+{
+    /// <summary>
+    /// A basic implementation of a document cache, limited by a configured amount of memory.
+    /// </summary>
+    public class MemoryDocumentCache : IDocumentCache, IDisposable
+    {
+        private readonly IMemoryCache _memoryCache;
+        private readonly MemoryDocumentCacheOptions _options;
+
+        /// <summary>
+        /// Initializes a new instance with the default options: 100,000 maximum total query size and no expiration time.
+        /// </summary>
+        public MemoryDocumentCache()
+            : this(new MemoryDocumentCacheOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance with the specified options.
+        /// </summary>
+        /// <param name="options">A value containing the <see cref="MemoryDocumentCacheOptions"/> to use.</param>
+        public MemoryDocumentCache(IOptions<MemoryDocumentCacheOptions> options)
+            : this(
+                new MemoryCache(new MemoryCacheOptions { SizeLimit = options.Value.MaxTotalQueryLength }),
+                options)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance with the specified memory cache and sliding expiration time period.
+        /// Note that by overriding <see cref="GetMemoryCacheEntryOptions(string)"/>, the sliding expiration
+        /// time specified within <paramref name="options"/> can be ignored.
+        /// </summary>
+        protected MemoryDocumentCache(IMemoryCache memoryCache, IOptions<MemoryDocumentCacheOptions> options)
+        {
+            _memoryCache = memoryCache ?? throw new ArgumentNullException(nameof(memoryCache));
+            _options = options?.Value;
```

Comment:
```suggestion
            _options = options?.Value ?? throw new ArgumentNullException(nameof(options));;
```

Reply:
Options is not necessary if they are overriding `GetMemoryCacheEntryOptions`

Reply:
Yeah, OK.

Reply:
But....

```c#
 _options = options?.Value ?? (GetType() == typeof(MemoryDocumentCache) ? null : throw new ArgumentNullException(nameof(options)));
```
😄